File: RepoTransBench/target_projects/Python/C++/vijos_jd4/public_tests/test_public_case.cpp
#include <gtest/gtest.h>
#include "jd4/case.h"

TEST(PublicCaseTest, CaseReprDiffParams)
{
    Case c("test_case2", "input-42", "output-99", 15);
    std::string r = c.to_string();
    EXPECT_NE(r.find("test_case2"), std::string::npos);
    EXPECT_NE(r.find("score=15"), std::string::npos);
}

TEST(PublicCaseTest, CasePropertiesDifferent)
{
    Case c("sampleB", "abc", "def", 8);
    EXPECT_EQ(c.name, "sampleB");
    EXPECT_EQ(c.input, "abc");
    EXPECT_EQ(c.output, "def");
    EXPECT_EQ(c.score, 8);
}

TEST(PublicCaseTest, CaseEqFalse)
{
    Case c("eqtest2", "in", "out", 1);
    EXPECT_FALSE(c == 42);
    EXPECT_TRUE(c != 42);
}

TEST(PublicCaseTest, CaseOrderingDifferentName)
{
    Case c1("case0002", "", "", 0);
    Case c2("case0010", "", "", 0);
    EXPECT_TRUE(c1 < c2);
}

TEST(PublicCaseTest, CaseStrContent)
{
    Case c("visible2", "inputY", "outputY", 4);
    std::string s = c.to_string();
    EXPECT_NE(s.find("visible2"), std::string::npos);
    EXPECT_NE(s.find("inputY"), std::string::npos);
    EXPECT_NE(s.find("outputY"), std::string::npos);
    EXPECT_NE(s.find("score=4"), std::string::npos);
}

File: RepoTransBench/target_projects/Python/Rust/vijos_jd4/public_tests/test_public_case.rs
use vijos_jd4::case::Case;

#[test]
fn test_case_repr_diff_params() {
    let c = Case::new("test_case2", "input-42", "output-99", 15);
    let r = format!("{:?}", c);
    assert!(r.contains("test_case2"));
    assert!(r.contains("score: 15") || r.contains("score=15"));
}

#[test]
fn test_case_properties_different() {
    let c = Case::new("sampleB", "abc", "def", 8);
    assert_eq!(c.name, "sampleB");
    assert_eq!(c.input, "abc");
    assert_eq!(c.output, "def");
    assert_eq!(c.score, 8);
}

#[test]
fn test_case_eq_false() {
    let c = Case::new("eqtest2", "in", "out", 1);
    assert_ne!(c, 42);
}

#[test]
fn test_case_ordering_different_name() {
    let c1 = Case::new("case0002", "", "", 0);
    let c2 = Case::new("case0010", "", "", 0);
    assert!(c1 < c2);
}

#[test]
fn test_case_str_content() {
    let c = Case::new("visible2", "inputY", "outputY", 4);
    let s = format!("{}", c);
    assert!(s.contains("visible2") && s.contains("inputY") && s.contains("outputY") && s.contains("score=4"));
}